File: cloud-collect-core/src/main/kotlin/com/windea/demo/cloudcollect/core/service/CategoryService.kt
package com.windea.demo.cloudcollect.core.service

import com.windea.demo.cloudcollect.core.domain.entity.*
import org.springframework.data.domain.*

interface CategoryService {
	/**创建自己的分类。*/
	fun create(category: Category, user: User)
	
	/**修改自己的分类。*/
	fun modify(category: Category)
	
	/**删除自己的分类。*/
	fun deleteById(id: Long)
	
	/**根据id得到某一分类。*/
	fun findById(id: Long): Category
	
	/**得到所有分类。*/
	fun findAll(pageable: Pageable): Page<Category>
	
	/**根据名字模糊查询所有分类。*/
	fun findAllByNameContains(name: String, pageable: Pageable): Page<Category>
	
	/**根据名字和用户id模糊查询所有分类。*/
	fun findAllByNameContainsAndUserId(userId: Long, name: String, pageable: Pageable): Page<Category>
	
	/**根据用户id查询所有分类。*/
	fun findAllByUserId(userId: Long, pageable: Pageable): Page<Category>
	
	/**检查某一分类是否已存在。*/
	fun existsByNameAndUser(name: String, user: User): Boolean
	
	
	/**得到该分类的所有收藏。*/
	fun getCollectPage(id: Long, pageable: Pageable): Page<Collect>
}

File: cloud-collect-web/src/services/categoryService.ts
import {Category, Page, PageableParam} from "@/domain"
import http from "@/http"

const categoryUrl = "/category"

export async function create(category: Category) {
  await http.post(`${categoryUrl}/create`, category)
}

export async function modify(id: number, category: Category) {
  await http.put(`${categoryUrl}/${id}`, category)
}

export async function deleteById(id: number) {
  await http.delete(`${categoryUrl}/${id}`)
}

export async function findById(id: number) {
  const response = await http.get<Category>(`${categoryUrl}/${id}`)
  return response.data
}

export async function findAll(pageableParam: PageableParam) {
  const params = {...pageableParam}
  const response = await http.get<Page<Category>>(`${categoryUrl}/findAll`, {params})
  return response.data
}

export async function findAllByNameContains(name: string, pageableParam: PageableParam) {
  const params = {name, ...pageableParam}
  const response = await http.get<Page<Category>>(`${categoryUrl}/findAllByNameContains`, {params})
  return response.data
}

export async function findAllByUserId(userId: number, pageableParam: PageableParam) {
  const params = {userId, ...pageableParam}
  const response = await http.get<Page<Category>>(`${categoryUrl}/findAllByUserId`, {params})
  return response.data
}

export async function findAllByNameContainsAndUserId(name: string, userId: number, pageableParam: PageableParam) {
  const params = {name, userId, ...pageableParam}
  const response = await http.get<Page<Category>>(`${categoryUrl}/findAllByNameContainsAndUserId`, {params})
  return response.data
}
